docs/api/selector.md

```diff
@@ -12,6 +12,16 @@ follows:
 - class syntax (`.foo`, `.foo-bar`, etc.)
 - tag syntax (`input`, `div`, `span`, etc.)
 - id syntax (`#foo`, `#foo-bar`, etc.)
+- attr syntax(`[foo="bar"]`, `[foo]`, etc.);
+
+A note on attribute selectors: For compatibilty, you can use either the
+react prop name, or the native html selector. For example, the following
```

Comment:
@ljharb I can understand your hesitation with the API decision. We solved something similar in the testing library [skin-deep](https://github.com/glenjamin/skin-deep/blob/master/test/test.js#L562) with `tree.subTreeLike('*', {foo: 'bar'})`.

We wanted to query for specific components using props, so we could use specific testing properties like `data-testref='foo'` that didn't interfere with our ui teams use of classnames.

cc @lelandrichardson @blainekasten